Signal/src/AppDelegate.m

```diff
@@ -630,38 +624,42 @@ - (BOOL)application:(UIApplication *)application
+                    return;
+                }
             }
-        }
 
-        // This intent can be received from more than one user interaction.
-        //
-        // * It can be received if the user taps the "video" button in the CallKit UI for an
-        //   an ongoing call.  If so, the correct response is to try to activate the local
-        //   video for that call.
-        // * It can be received if the user taps the "video" button for a contact in the
-        //   contacts app.  If so, the correct response is to try to initiate a new call
-        //   to that user - unless there already is another call in progress.
-        if (SignalApp.sharedApp.callService.call != nil) {
-            if ([phoneNumber isEqualToString:SignalApp.sharedApp.callService.call.remotePhoneNumber]) {
-                DDLogWarn(@"%@ trying to upgrade ongoing call to video.", self.logTag);
-                [SignalApp.sharedApp.callService handleCallKitStartVideo];
-                return YES;
-            } else {
-                DDLogWarn(
-                    @"%@ ignoring INStartVideoCallIntent due to ongoing WebRTC call with another party.", self.logTag);
-                return NO;
+            // This intent can be received from more than one user interaction.
+            //
+            // * It can be received if the user taps the "video" button in the CallKit UI for an
+            //   an ongoing call.  If so, the correct response is to try to activate the local
+            //   video for that call.
+            // * It can be received if the user taps the "video" button for a contact in the
+            //   contacts app.  If so, the correct response is to try to initiate a new call
+            //   to that user - unless there already is another call in progress.
+            if (SignalApp.sharedApp.callService.call != nil) {
+                if ([phoneNumber isEqualToString:SignalApp.sharedApp.callService.call.remotePhoneNumber]) {
+                    DDLogWarn(@"%@ trying to upgrade ongoing call to video.", self.logTag);
+                    [SignalApp.sharedApp.callService handleCallKitStartVideo];
+                    return;
+                } else {
+                    DDLogWarn(@"%@ ignoring INStartVideoCallIntent due to ongoing WebRTC call with another party.",
+                        self.logTag);
+                    return;
+                }
             }
-        }
 
-        OutboundCallInitiator *outboundCallInitiator = SignalApp.sharedApp.outboundCallInitiator;
-        OWSAssert(outboundCallInitiator);
-        return [outboundCallInitiator initiateCallWithHandle:phoneNumber];
+            OutboundCallInitiator *outboundCallInitiator = SignalApp.sharedApp.outboundCallInitiator;
+            OWSAssert(outboundCallInitiator);
+            [outboundCallInitiator initiateCallWithHandle:phoneNumber];
+        }];
+        return YES;
```

Comment:
I'm unfamiliar with the ramifications of always returning YES here.

Reply:
I've added a comment.